src/app/add-item/add-item.component.ts

```diff
@@ -0,0 +1,56 @@
+import { TasksService, ActionItem } from './../services/tasksService/tasks.service';
+import { Project, ProjectsService } from '../services/projects/projects.service';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Validators, FormGroup, FormBuilder } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-add-item',
+  templateUrl: './add-item.component.html',
+  styleUrls: ['./add-item.component.css']
+})
+export class AddItemComponent implements OnInit, OnDestroy {
+  dialogForm: FormGroup;
+  title = 'Create Action Item';
+  projects: Project[];
+  newActionItems: ActionItem[] = [];
+  buttonSpinner = false;
```

Comment:
It is better to use more business specific names for variables. So for example here it would be better to replace that to something like `isCreatingActionItem` or something like that. especially @odudycz pay a lot of attention to this aspect ;)